### PR TITLE
`Development`: Delay scheduled task to speed up application start

### PR DESCRIPTION
**Pull request by krusche**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

### Motivation and Context
Application start can be slow if many tasks need to be scheduled. This PR delays this until shortly after the startup of the server application is finished. 

### Testing
Start the application and make sure tasks are scheduled shortly after the server application started

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/scheduled/ProgrammingExerciseScheduleService.java [hunk 1]
```diff
@@ -936,7 +951,7 @@ private CompletableFuture<List<ProgrammingExerciseStudentParticipation>> stashCh
     private CompletableFuture<List<ProgrammingExerciseStudentParticipation>> invokeOperationOnAllParticipationsThatSatisfy(Long programmingExerciseId,
             BiConsumer<ProgrammingExercise, ProgrammingExerciseStudentParticipation> operation, Predicate<ProgrammingExerciseStudentParticipation> condition,
             String operationName) {
-        log.info("Invoking (scheduled) task '{}' for programming exercise with id {}.", operationName, programmingExerciseId);
+        log.debug("Invoking (scheduled) task '{}' for programming exercise with id {}.", operationName, programmingExerciseId);
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/service/scheduled/ProgrammingExerciseScheduleService.java:954]: imo this should stay at info for checking if a task actually ran.

**Review by nityanandaz — APPROVED** [review]
Manual local test: Scheduler calls service's, e.g., ModelingExerciseScheduleService, method after 25 seconds, as expected.

**Review by florian-glombik — COMMENTED** [review]
The adjusted services `ModelingExerciseScheduleService`, `NotificationScheduleService`, `ProgrammingExerciseScheduleService`, and `WeeklyEmailSummaryScheduleService` are started after ~25 seconds

I would have interpreted the server to be started as "up and running", in other words, reachable from outside (or on port 8080 if hosted locally).
On my local machine, the startup takes 26-27 seconds, which means that the delay is not sufficient to make sure that the schedule services do not delay the startup.

<details><summary>Logs from local testing</summary>

```
:: Interactive Learning with Individual Feedback ::

:: Powered by Spring Boot 3.2.5 ::

2024-05-11T22:24:37.980+02:00 DEBUG 4393 --- [Artemis] [kground-preinit] org.jboss.logging                        : Logging Provider: org.jboss.logging.Slf4jLoggerProvider found via system property
...
2024-05-11T22:25:04.344+02:00 DEBUG 4393 --- [Artemis] [           main] d.t.i.w.a.aop.logging.LoggingAspect      : Enter: de.tum.in.www1.artemis.service.connectors.localci.buildagent.LocalCIDockerService.applicationReady() with argument[s] = []
2024-05-11T22:25:04.344+02:00 DEBUG 4393 --- [Artemis] [           main] d.t.i.w.a.aop.logging.LoggingAspect      : Exit: de.tum.in.www1.artemis.service.connectors.localci.buildagent.LocalCIDockerService.applicationReady() with result = null
2024-05-11T22:25:04.344+02:00 DEBUG 4393 --- [Artemis] [           main] d.t.i.w.a.aop.logging.LoggingAspect      : Enter: de.tum.in.www1.artemis.service.scheduled.ModelingExerciseScheduleService.applicationReady() with argument[s] = []
2024-05-11T22:25:04.344+02:00 DEBUG 4393 --- [Artemis] [           main] d.t.i.w.a.aop.logging.LoggingAspect      : Exit: de.tum.in.www1.artemis.service.scheduled.ModelingExerciseScheduleService.applicationReady() with result = null
2024-05-11T22:25:04.345+02:00 DEBUG 4393 --- [Artemis] [           main] d.t.i.w.a.aop.logging.LoggingAspect      : Enter: de.tum.in.www1.artemis.service.scheduled.NotificationScheduleService.applicationReady() with argument[s] = []
2024-05-11T22:25:04.345+02:00 DEBUG 4393 --- [Artemis] [           main] d.t.i.w.a.aop.logging.LoggingAspect      : Exit: de.tum.in.www1.artemis.service.scheduled.NotificationScheduleService.applicationReady() with result = null
2024-05-11T22:25:04.345+02:00 DEBUG 4393 --- [Artemis] [           main] d.t.i.w.a.aop.logging.LoggingAspect      : Enter: de.tum.in.www1.artemis.service.scheduled.ProgrammingExerciseScheduleService.applicationReady() with argument[s] = []
2024-05-11T22:25:04.345+02:00 DEBUG 4393 --- [Artemis] [           main] d.t.i.w.a.aop.logging.LoggingAspect      : Exit: de.tum.in.www1.artemis.service.scheduled.ProgrammingExerciseScheduleService.applicationReady() with result = null
2024-05-11T22:25:04.345+02:00 DEBUG 4393 --- [Artemis] [           main] d.t.i.w.a.aop.logging.LoggingAspect      : Enter: de.tum.in.www1.artemis.service.scheduled.WeeklyEmailSummaryScheduleService.applicationReady() with argument[s] = []
2024-05-11T22:25:04.345+02:00 DEBUG 4393 --- [Artemis] [           main] d.t.i.w.a.aop.logging.LoggingAspect      : Exit: de.tum.in.www1.artemis.service.scheduled.WeeklyEmailSummaryScheduleService.applicationReady() with result = null
2024-05-11T22:25:04.345+02:00 DEBUG 4393 --- [Artemis] [           main] d.t.i.w.a.aop.logging.LoggingAspect      : Enter: de.tum.in.www1.artemis.service.scheduled.cache.quiz.QuizScheduleService.applicationReady() with argument[s] = []
2024-05-11T22:25:04.346+02:00 DEBUG 4393 --- [Artemis] [           main] d.t.i.w.a.c.ArtemisSpringManagedContext  : Initialize obj de.tum.in.www1.artemis.service.scheduled.cache.quiz.QuizProcessCacheTask@481e616f of type de.tum.in.www1.artemis.service.scheduled.cache.quiz.QuizProcessCacheTask
2024-05-11T22:25:04.352+02:00 DEBUG 4393 --- [Artemis] [cached.thread-5] d.t.i.w.a.aop.logging.LoggingAspect      : Enter: de.tum.in.www1.artemis.service.scheduled.cache.quiz.QuizScheduleService.processCachedQuizSubmissions() with argument[s] = []
...
2024-05-11T22:25:04.640+02:00  INFO 4393 --- [Artemis] [           main] de.tum.in.www1.artemis.ArtemisApp        :
----------------------------------------------------------
	'Artemis' is running! Access URLs:
	Local:      http://localhost:8080/
	External:   http://192.168.178.31:8080/
	Profiles:   [dev, localci, localvc, artemis, scheduling, buildagent, core, ldap-only, local]
	Version:    7.0.5
	Git Commit: da1459a
	Git Branch: chore/delay-scheduled-tasks-on-startup
----------------------------------------------------------
```

</details>

File: src/main/java/de/tum/in/www1/artemis/service/scheduled/cache/quiz/QuizScheduleService.java [hunk 2]
```diff
@@ -293,17 +293,17 @@ public void startSchedule(long delayInMillis) {
     public void stopSchedule() {
```
Comment (rstief) [inline src/main/java/de/tum/in/www1/artemis/service/scheduled/cache/quiz/QuizScheduleService.java:293]: As far as I can tell the QuizScheduleService gets created on application start (see line 116.) Is this intentional or should it also be delayed?

**Review by rstief — DISMISSED** [review]
Code looks good 👍 Just one small question

File: src/main/java/de/tum/in/www1/artemis/service/scheduled/cache/quiz/QuizScheduleService.java (hunk 2 above)
Comment (krusche) [reply src/main/java/de/tum/in/www1/artemis/service/scheduled/cache/quiz/QuizScheduleService.java:299]: good point, I delayed it as well

**Review by julian-christl — DISMISSED** [review]
code

**Pull request merged**